Add start and end vertex label ids and names in edge object

Zainab-Saad opened the pull request:
A sample of modified edge object is 
```
{"id": 1, "label": "PARTICIPATES", "end_id": 2, "start_id": 1, "properties": {"from": 2021}, "end_label_id": 9, "end_label_name": "Project", "start_label_id": 3, "start_label_name": "Developer"}
```
14 of 26 regression tests fail after these changes (none have been fixed in this PR yet)

Zainab-Saad committed:
Add start and end vertex label ids and names in edge object

code changes done to reflect this change are:

 * Modify the SQL function -  _agtype_build_edge to include start and end vertex
   label ids and names in edge object

 * Modify executor, age_vle and edge typecasting to pass label id and name
   when constructing the edge object or retrieve these values while unpacking
   an edge

 * Update MATCH to work without using _extract_label_id (removed this function)

Update the regression tests

rafsun42 commented:
@Zainab-Saad Thank you.

Could you update the test files that were affected by this PR please?

Zainab-Saad commented:
@rafsun42 
I have fixed the failing regression tests and done some code changes mainly for:
- making MATCH work without using `_extract_label_id` (removed it)
- Updated `age_vle` and edge/path `typecasting` to reflect the changes in edge object

Now 3 of 26 tests fail (`expr, age_load, graph_generation`) which will be fixed by PR # 26 and 29

rafsun42 commented:
@Zainab-Saad 

Looks good. However- 

1. `graph_generation` has errors `ERROR:  graph_oid and label_id must not be null`.
2. `age_load` also has errors. But I assume the errors will be fixed in age_load's PR. Is that correct?

rafsun42 commented on regress/expected/agtype.out:
```diff
@@ -1629,32 +1629,14 @@ NOTICE:  graph "orderability_graph" has been created
 ---
 (0 rows)
 
-SELECT * FROM cypher('orderability_graph', $$ MATCH (n) RETURN n ORDER BY n.prop $$) AS (sorted agtype);
```
Is this change expected?

rafsun42 commented on regress/expected/agtype.out:
```diff
@@ -1629,32 +1629,14 @@ NOTICE:  graph "orderability_graph" has been created
 ---
 (0 rows)
 
-SELECT * FROM cypher('orderability_graph', $$ MATCH (n) RETURN n ORDER BY n.prop $$) AS (sorted agtype);
```
Why is this test removed?

Zainab-Saad replied:
it's a mistake, I have fixed that

rafsun42 replied:
@Zainab-Saad Okay. Are you sure that rest of the tests are correct?

Zainab-Saad commented:
> 1. graph_generation has errors ERROR:  graph_oid and label_id must not be null. 2. age_load also has errors. But I assume the errors will be fixed in age_load's PR. Is that correct?

Yes, these errors will be fixed with PR on age_load
